site/content/blog/2019-01-20-submitting-great-prs.md

```diff
@@ -0,0 +1,62 @@
+
+I am forever grateful to my mentor for shaping my thinking about reviewing code. Having given more than a few reviews since then, let me share some advice, starting with what my mentor taught me:
+
+### 1) Review your own code first.
+
+Be the one to catch styling issues. Be the one to ask, "does this variable name make any sense?" Be the one to identify problem areas and points of confusion. Be the first reviewer.
+
+Now, not everything deserves to be fixed in every pull request. I love to refactor code, but there's a time and a place and fixing a one-line bug isn't usually the time to fix a multi-file, variable naming issue. This leads to my second point:
+
+### 2) Give a great description.
+
+The description on a pull request is your chance to prove to a reviewer that you have the credentials to fix this problem. How? State the problem in your own words:
+
+> This pull request adds new functionality to expose ...
+
+Tell the reviewer why we need to do this:
+
+> With this new functionality, we can fix a longstanding customer issue that ...
+
+You don't need to outline everything your code is doing -- that's what the code is there for -- but this is also your opportunity to point the reviewer in the right direction:
+
+> The trickiest part of this pull request was the `SqlStore` changes, because ...
+
+and also:
+
+> I really wanted to fix the variable names in this file, but it's out of scope for this pull request.
+
+Of course, sometimes you have to fix something else before you can get to the issue at hand. Maybe you decide to bite the bullet and fix that variable name problem because it clarifies the feature you're adding. On to my third point:
+
+### 3) Clean up your history.
+
+Everyone has their own development style. Mine usually consists of mashing the keyboard, then typing:
+
+    git commit -m wip
+
+and repeating this process. Commit early and commit often! It's just easier to undo changes this way while you're actively developing.
+
+But a reviewer doesn't need to see that commit history. As a reviewer myself, I'd want to see one of two things. Either a single squashed commit with all of your changes, or a history breaking up your changes into logical groupings. Put that variable name fix in its own commit, commit your feature change on top of it, and add a note to your PR's description. Your reviewers can then use the commits tab in GitHub to review those changes independently.
+
+Maybe you're the kind of developer that writes a pristine commit history the first time, but most of us rely on `git rebase`. If you aren't familiar with rebasing, it might seem terrifying or magical, but it doesn't need to be: play with it, master it, and use it often. Hands down, it's my favourite feature in git. If you need help getting started here, check out Atlassian's excellent [tutorial on git rebase](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
```

Comment:
`hands down` translates to `lower your hand`in Japanese or `handing on to the next generation` in Finnish.